uptime_service_validation/coordinator/coordinator.py

```diff
@@ -18,294 +18,309 @@
-                                "blockchain_height",
-                                "amount",
-                                "created_at",
-                                "bot_log_id",
-                                "state_hash",
-                            ]
+                    point_record_df.loc[:, "bot_log_id"] = bot_log_id
+                    point_record_df = point_record_df[
+                        [
+                            "file_name",
+                            "file_timestamps",
+                            "blockchain_epoch",
+                            "block_producer_key",
+                            "blockchain_height",
+                            "amount",
+                            "created_at",
+                            "bot_log_id",
+                            "state_hash",
                         ]
+                    ]
 
-                        createPointRecord(connection, logging, point_record_df)
-                except Exception as error:
-                    connection.rollback()
-                    logging.error(ERROR.format(error))
-                finally:
-                    connection.commit()
-                process_loop_count += 1
-                logging.info("Processed it loop count : {0}".format(process_loop_count))
-
-            else:
-                logging.info("Finished processing data from table.")
-            try:
-                updateScoreboard(
-                    connection,
-                    logging,
-                    cur_batch_end,
-                    int(os.environ["UPTIME_DAYS_FOR_SCORE"]),
-                )
+                    createPointRecord(connection, logging, point_record_df)
             except Exception as error:
                 connection.rollback()
+                bot_log_id = None
                 logging.error(ERROR.format(error))
-            finally:
+            else:
                 connection.commit()
 
-            prev_batch_end = cur_batch_end
-            cur_batch_end = prev_batch_end + timedelta(minutes=interval)
-            if prev_batch_end >= cur_timestamp:
-                logging.warning(
-                    "It seems that batch processing took a bit too long than expected as prev_batch_end: %s >= cur_timestamp: %s... progressing to the next batch anyway...",
-                    prev_batch_end,
-                    cur_timestamp,
-                )
+        else:
+            logging.info("Finished processing data from table.")
+        try:
+            updateScoreboard(
+                connection,
+                logging,
+                cur_batch_end,
+                int(os.environ["UPTIME_DAYS_FOR_SCORE"]),
+            )
+        except Exception as error:
+            connection.rollback()
+            logging.error(ERROR.format(error))
+        else:
+            connection.commit()
+        return bot_log_id
+
+
+def main():
+    process_loop_count = 0
+    retry_count = 0
```

Comment:
I'm thinking it might be actually useful to make it configurable as env variable. It may be useful for tweaking on the deployment environment.

Reply:
Like this: 85d40ad057cd3f4222c66b2aaa122246adf54002?